interface/patient_file/report/patient_report.php

```diff
@@ -21,6 +46,94 @@
   $ptdata = getPatientData($pid, 'cmsportal_login');
   $cmsportal = $ptdata['cmsportal_login'] !== '';
 }
+
+// May get directives from GET or POST
+$pauto = ($_SERVER['REQUEST_METHOD']=='GET' ? $_GET : $_POST);
+
+/* Expected directives are:
+ *  pid  = <patient id to override current $pid>
+ *  enc  = <limited to this encounter>
+ *  auto = [y]es for automated actions
+ *  sel  = [y]es to select checkboxes but do not generate the report
+ *  fee  = [y]es, [n]o
+ *  note = [y]es, [n]o
+ *  comm = [y]es, [n]o
+ *  mths = <number of months>
+*/
+
+foreach ($pauto as $key => $val) {
+  switch ($key) {
+    case 'pid':
+      setpid($pauto['pid']);
+      break;
+    case 'enc':
+      setencounter($pauto['enc']);
+      break;
+    case 'mths':
+      $pauto['dt_from'] = date("Y-m-d", strtotime("-".$pauto['mths']." months"));
+      $pauto['dt_to'] = date("Y-m-d");
+      break;
+    case 'sel':
+    case 'fee':
+    case 'note':
+    case 'comm':
+      if (substr($pauto[$key], 0, 1) != 'y') unset($pauto[$key]);
+      break;
+      
+    default:
+      // Skip unrecognized parameter
+    break;
+  }
+}
+
+$opt_chk = array(
+  'fee' => 'opt_fee',
+  'note'=> 'opt_note',
+  'comm'=> 'opt_comm'
+);
+
+$auto_js = '';
+$disp_none = false;
+if ($pauto['auto']) {
+  $auto_js = 'checkAll(true);';
+  foreach ($opt_chk as $key => $val) {
+    if (!isset($pauto[$key])) {
+      $auto_js .= "$('#$val').html('');";
+    }
+  }
+  if (!isset($pauto['sel'])) {
+    $auto_js .= 'formSubmit();';
+    $disp_none = true;
+  } 
+}
+$sql_enc = "SELECT f.encounter, f.form_id, f.form_name, f.formdir, f.date AS fdate, fe.date, fe.reason 
+    FROM forms f INNER JOIN form_encounter fe ON fe.pid = f.pid AND fe.encounter = f.encounter
+    WHERE f.pid=? AND f.deleted=0";
+$fil_enc = array($pid);
+$sql_po = "SELECT po.procedure_order_id, po.date_ordered, fe.date 
+    FROM procedure_order po LEFT JOIN forms f ON f.pid = po.patient_id AND f.form_id = po.procedure_order_id  
+    LEFT JOIN form_encounter fe ON fe.pid = f.pid AND fe.encounter = f.encounter 
+    WHERE po.patient_id = ? AND f.formdir = 'procedure_order' AND f.deleted = 0";
```

Comment:
I think that approach will run into trouble on a left join where there is no match, because the where conditions will be comparing against null values.  I have always used the ON clause to specify the conditions for its related join.  In this case it will work but in general I think it makes things more complicated.  Might be interesting to look up accepted best practice.